test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.test.junit.nativeimage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Locale;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * This is a general utility to assert via
+ * unit testing which classes have been included in a native-image.
+ */
+public final class ClassInclusionReport {
```

Comment:
Having this might enable tracking noticeable class inclusion increases over different versions?

Reply:
yes exactly! Both caused by changes in our code and changes in GraalVM - it will be tested on each build, we just need to use this test helper more frequently, and make sensible choices regarding which types we consider the contract broken.